### PR TITLE
set static UUID for spectators

### PR DESCRIPTION
**Pull request by bradcarnage**
disconnect spectators on duplicate UUID or Username
if else cleanup

### CODE REVIEW

**Comment by bradcarnage** [comment]
apologies for the addl commits. forgot to re-fork after #39

**Comment by rfresh2** [comment]
> apologies for the addl commits. forgot to re-fork after #39

You should create a new branch for every PR. That will avoid any issues in the future

File: src/main/java/com/zenith/network/server/handler/ProxyServerLoginHandler.java [hunk 1]
```diff
@@ -76,35 +76,37 @@ && nonNull(CACHE.getTabListCache().getTabList().get(CACHE.getProfileCache().getP
                     CACHE.getPlayerCache().getPortalCooldown()
             ));
             if (!proxy.isInQueue()) { PlayerCache.sync(); }
+        } else if (clientGameProfile.getId().equals(CACHE.getProfileCache().getProfile().getId())) {
+            connection.disconnect("You can't spectate your own UUID.");
+        } else if (clientGameProfile.getName().equals(CACHE.getProfileCache().getProfile().getName())) {
+            connection.disconnect("You can't spectate your own Username.");
```
Comment (rfresh2) [inline src/main/java/com/zenith/network/server/handler/ProxyServerLoginHandler.java]: There's gotta be better solutions than blocking this use case entirely. e.g. change the spectator's game profile at login

**Comment by bradcarnage** [comment]
fixes the issue(s) where you can't see the profile you're already logged in as in the game.

File: src/main/java/com/zenith/network/server/handler/ProxyServerLoginHandler.java (hunk 1 above)
Comment (bradcarnage) [reply src/main/java/com/zenith/network/server/handler/ProxyServerLoginHandler.java]: > There's gotta be better solutions than blocking this use case entirely. e.g. change the spectator's game profile at login

i just wasn't smart enough. :)

**Pull request merged**